### DISCUSSION TITLE
How to run a script before and after backup?

### DISCUSSION THREAD

**Discussion by Zalfsten** [vmware-tanzu/velero#8226] [Community support Q&A]
Hello,

I am trying to make a backup of my nextcloud instance with velero.

The official backup procedure for nextcloud is:

* activate the nextcloud maintenance mode
* perform a filesystem backup
* perform a database backup
* exit maintenance mode

So I would like to run scripts before and after the velero backup process.
As I understand it, this does not work with pre/post backup hooks, because they do not run before and after the actual backup process, but before and after each individual pod backup. So I could implement a procedure like this:

* activate maintenance mode
* perform a filesystem backup
* Exit maintenance mode
* possibly many additional backup steps of other kubernetes objects
* activate maintenance mode
* perform a database backup
* Exit maintenance mode

Even this is not trivial as the database pod would need to talk to the nextcloud pod to enter maintenance mode.
Nevertheless this could lead to inconsistent backups, of course.

Is there a solution to map the official nextcloud backup process with velero?

Thanks and Regards,
Carsten

**Comment by Zalfsten** [comment]
Seems to be related to #3978

**Reply by kaovilai** [reply]
also https://github.com/vmware-tanzu/velero/issues/4435
